Source/Core/Common/MemoryUtil.cpp

```diff
@@ -46,10 +49,15 @@ void* AllocateExecutableMemory(size_t size, void* map_hint)
 		PanicAlert("Failed to allocate executable memory.");
 	}
 
-#ifdef _X86_64
-	ptrdiff_t ofs = (u8*)ptr - (u8*)map_hint;
-	if (ofs < -0x80000000ll || ofs + size > 0x80000000ll)
-		PanicAlert("Executable range can't be used for RIP-relative addressing.");
+#ifdef _M_X86_64
+	if (map_hint)
+	{
+		ptrdiff_t ofs = (u8*)ptr - (u8*)map_hint;
+		if (ofs < -0x80000000ll || ofs + (ptrdiff_t)size > 0x80000000ll)
+			PanicAlert("%p can't be used for RIP-relative addressing. "
```

Comment:
This check is definitely not right.  For one thing, it doesn't cover the whole region.  For another, aren't we supposed to be getting full position independence? - if not, the check should be in whatever allocations actually care about this, not the mmap function.